src/base/anim/control.cpp

```diff
@@ -150,15 +143,48 @@ void Control::finish(bool preRun)
 		}
 	}
 	else if (preRun
-	         && ((direction == Direction::normal && atEndPosition())
-	             || (direction == Direction::reverse
+	         && ((options.direction == Direction::normal
+	                 && atEndPosition())
+	             || (options.direction == Direction::reverse
 	                 && atStartPosition()))
-	         && playState != PlayState::running) {
+	         && options.playState != PlayState::running) {
 		if (!finished && onFinish) {
 			onFinish(true);
 			finished = true;
 		}
 	}
 }
 
+void Control::setValue(std::string_view path,
```

Comment:
This code was on the interface because reflecting method/params of a class shouldn't be the responsibility of the actual class.

Reply:
OK, I moved back to the interface.